packages/user-group-spa/src/components/Home.jsx

```diff
@@ -0,0 +1,168 @@
+import React, { useState, useEffect } from 'react';
+import { Link, useHistory } from 'react-router-dom';
+import {
+  Toolbar,
+  ToolbarContent,
+  ToolbarItem,
+  SearchInput,
+  Button,
+  EmptyState,
+  EmptyStateIcon,
+  EmptyStateBody,
+  EmptyStatePrimary,
+  Spinner,
+  Title,
+  Pagination,
+} from '@patternfly/react-core';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+} from '@patternfly/react-table';
+import GroupsAPI from '../services/GroupsAPI';
+
+function Home ( props ) {
+  let [ searchText, setSearchText ] = useState( '' );
+  let [ loading, setLoading ] = useState( true );
+  let [ groups, setGroups ] = useState( [] );
+  let [ searchResults, setSearchResults ] = useState( [] );
+  let [ pagination, setPagination ] = useState( { page: 1, perPage: 10 } );
```

Comment:
Use const instead of let.

Reply:
It has to be `let` because those are state hooks, which means they are redefined every time its value changes.

Reply:
No, you must never modify state directly. you should always use setter returned by the hook.